project/tickets/views.py

```diff
@@ -0,0 +1,312 @@
+                new_ticket = ticket_form.save()
+                new_ticket_id = new_ticket.pk
+                messages.success(
+                    request, f"SUCCESS! Ticket for a FEATURE created!\
+                        €{donation_amount} was charged to your card.")
+                return redirect(tickets_view_one, new_ticket_id)
+            else:
+                messages.error(request, f"Unable to take payment!")
+        else:
+            print(ticket_form.errors)
+            messages.error(request, f"There was an error. Please try again.")
+    else:
+        ticket_form = TicketForm()
+        donation_form = DonationForm()
+    context = {
+        "donation_form": donation_form,
+        "ticket_form": ticket_form,
+        "publishable": settings.STRIPE_PUBLISHABLE
+    }
+    return render(request, "tickets_new_feature.html", context)
+
+
+def tickets_view_one(request, pk):
+    """ View a Single Ticket """
+    # get ticket details
+    ticket = get_object_or_404(Ticket, pk=pk)
+    # increase number of views by 1
+    ticket.views += 1
+    ticket.save()
+    # filter comments on specific ticket
+    comments = Comment.objects.filter(ticket_id=ticket.pk)
+    # filter upvotes on specific ticket by user ID
+    upvotes = Upvote.objects.filter(ticket_id=ticket.pk).values("user_id")
+    voters = [vote["user_id"] for vote in upvotes]
+    donors = User.objects.filter(id__in=voters)
+    # render drop-down list for superusers to update status
+    ticket_status_list = TicketStatus.objects.all()
+    tkt_status = ticket.ticket_status.id
+    # POST methods
+    if request.method == "POST":
+        comment_form = CommentForm(request.POST)
+        if comment_form.is_valid():
+            comment_form.instance.commenter = request.user
+            comment_form.instance.ticket = ticket
+            comment_form.save()
+            # remove two ticket views to avoid duplicates on POST
+            ticket.views -= 2
+            ticket.save()
+            messages.success(
+                request, f"Your comment was added!")
+            return redirect(tickets_view_one, ticket.pk)
+    else:
+        comment_form = CommentForm()
+        donation_form = DonationForm()
+    context = {
+        "comment_form": comment_form,
+        "comments": comments,
+        "donation_form": donation_form,
+        "donors": donors,
+        "publishable": settings.STRIPE_PUBLISHABLE,
+        "ticket": ticket,
+        "ticket_status_list": ticket_status_list,
+        "tkt_status": tkt_status,
+        "voters": voters,
+    }
+    return render(request, "tickets_view_one.html", context)
+
+
+@login_required
+def tickets_edit(request, pk):
+    """ Edit a Single Ticket """
+    ticket = get_object_or_404(Ticket, pk=pk)
+    if request.method == "POST":
+        ticket_form = TicketForm(request.POST, instance=ticket)
+        if ticket_form.is_valid():
+            ticket_form.instance.date_edited = timezone.now()
```

Comment:
you shouldn't need this. it should be defined in your model

Reply:
It's in my model, but if I remove this line, it will not update the date_edited instance. I've just tested and nothing updates, so I will keep this in here for now.